src/api/python/_api.py

```diff
@@ -165,32 +237,60 @@ def register_memory(
     ) -> nixlBind.nixlRegDList:
         reg_descs = self.get_reg_descs(reg_list, mem_type, is_sorted)
 
-        # based on backend type and mem_type, figure what registrations are meaningful
         handle_list = []
         for backend_string in backends:
             handle_list.append(self.backends[backend_string])
         self.agent.registerMem(reg_descs, handle_list)
 
         return reg_descs
 
-    # The output from get_reg_descs (which is later passed to register_memory for
-    # registration) or direct output of register_memory is passed here
+    """
+    @brief Deregister memory regions from the specified backends.
+
+    @param dereg_list nixlRegDList of memory to deregister, from register_memory or get_reg_descs.
+    @param backends Optional list of backend names for deregistration, otherwise NIXL will deregister
+            with all the backends that have these memory regions registered.
+    """
+
     def deregister_memory(
         self, dereg_list: nixlBind.nixlRegDList, backends: list[str] = []
     ):
-        # based on backend type and mem_type, figure what deregistrations are needed
         handle_list = []
         for backend_string in backends:
             handle_list.append(self.backends[backend_string])
         self.agent.deregisterMem(dereg_list, handle_list)
 
-    # Optional proactive make connection
+    """
+    @brief Optional, to proactively establish a connection with a remote agent,
```

Comment:
We can add the rationale for this call being optional, but let's remove the leading term.